lib/archethic/contracts/interpreter/scope.ex

```diff
@@ -114,6 +114,32 @@ defmodule Archethic.Contracts.Interpreter.Scope do
     )
   end
 
+  def get_function_ast(function_name, nil) do
+    function_key = function_to_function_key(function_name, [])
+    get_in(Process.get(:scope), ["functions", function_key, :ast])
+  end
+
+  def get_function_ast(function_name, args) do
+    function_key = function_to_function_key(function_name, args)
+    get_in(Process.get(:scope), ["functions", function_key, :ast])
+  end
+
+  def execute_function_ast(function_name, args) do
+    get_function_ast(function_name, args)
+    |> Code.eval_quoted()
+    |> elem(0)
+  end
+
+  def function_to_function_key(function_name, args) do
```

Comment:
I think this function could be deleted since it is confusing to use it as we need to import the Scope module in other module not related to contract execution.
An other way to store function name and arity on the map key could be to use a tuple:
```elixir
%{
  {"function_name", 3} => %{ast: ast, args: args}
}
```
using this tuple could be also more easy to manipulate it if we need to get all arity available for a function name for example